dot/peerset/peerstate.go

```diff
@@ -258,17 +260,19 @@ func (ps *PeersState) addNoSlotNode(idx int, peerID peer.ID) {
 	}
 
 	ps.nodes[peerID] = n
+	return nil
 }
 
-func (ps *PeersState) removeNoSlotNode(idx int, peerID peer.ID) {
+func (ps *PeersState) removeNoSlotNode(idx int, peerID peer.ID) error {
 	if _, ok := ps.sets[idx].noSlotNodes[peerID]; !ok {
-		return
+		logger.Debugf("peer %s is not in no-slot node map", peerID)
+		return nil
 	}
 
 	delete(ps.sets[idx].noSlotNodes, peerID)
 	n, err := ps.getNode(peerID)
 	if err != nil {
-		return
+		return fmt.Errorf("could not get node: %w", err)
```

Comment:
just to keep consistent with the error in `addNoSlotNode` function 
```suggestion
		return fmt.Errorf("could not get node for peer id %s: %w", peerID, err)
```